<launch>
  
  <!-- node parameters -->
  <arg name="lidar_frame" default="base_link" />
  <arg name="map_frame" default="base_link" />
  <arg name="grid_min_value" default="0.0" />
  <arg name="grid_max_value" default="1.0" />
  <arg name="grid_resolution" default="0.25" /> <!-- 0.2, 0.25 -->
  <arg name="grid_length_x" default="40" />
  <arg name="grid_length_y" default="20" />
  <arg name="grid_position_x" default="18.5" />
  <arg name="grid_position_y" default="0" />
  <arg name="maximum_lidar_height_thres" default="-0.5" />
  <arg name="minimum_lidar_height_thres" default="-1.2" />

  <arg name="expand_polygon_size" default="-1.0" />
  <arg name="visual_expand_size" default="1.0" />
  <arg name="high_res_visual_expand_size" default="0.0" />
  <arg name="size_of_expansion_kernel" default="0" />
  <arg name="use_points" default="true" />
  <arg name="use_laserscan" default="false" />
  <arg name="use_wayarea" default="false" />
  <arg name="points_input" default="/os_cloud_node/below_points" /> 
  <arg name="scan_topic" default="/scan" />
  <arg name="enable_potential" default = "true" />
  <arg name="output_topic_name" default="/semantics/costmap_generator/occupancy_grid" />

  <!-- Launch node -->
  <node pkg="hyundai_costmap_generator_v2" type="hyundai_costmap_generator_v2" name="hyundai_costmap_generator_v2" output="screen">
    <param name="lidar_frame" value="$(arg lidar_frame)" />
    <param name="map_frame" value="$(arg map_frame)" />
    <param name="grid_min_value" value="$(arg grid_min_value)" />
    <param name="grid_max_value" value="$(arg grid_max_value)" />
    <param name="grid_resolution" value="$(arg grid_resolution)" />
    <param name="grid_length_x" value="$(arg grid_length_x)" />
    <param name="grid_length_y" value="$(arg grid_length_y)" />
    <param name="grid_position_x" value="$(arg grid_position_x)" />
    <param name="grid_position_y" value="$(arg grid_position_y)" />
    <param name="maximum_lidar_height_thres" value="$(arg maximum_lidar_height_thres)" />
    <param name="minimum_lidar_height_thres" value="$(arg minimum_lidar_height_thres)" />
    <param name="expand_polygon_size" value="$(arg expand_polygon_size)" />
    <param name="size_of_expansion_kernel" value="$(arg size_of_expansion_kernel)" />
    <param name="visual_expand_size" value="$(arg visual_expand_size)" />
    <param name="use_points" value="$(arg use_points)" />
    <param name="use_laserscan" value="$(arg use_laserscan)" />
    <param name="use_wayarea" value="$(arg use_wayarea)" />
    <param name="scan_topic" value="$(arg scan_topic)" />
    <param name="enable_potential" value="$(arg enable_potential)" />
    <param name="map_topic_name"   value="$(arg output_topic_name)" />
    <param name="points_topic_name"   value="$(arg points_input)" />
  </node>

</launch>
